### ISSUE TITLE
slope.py Vector*Matrix not supported B4.1 Stable

### ISSUE THREAD

**Issue by EvanDSanders** [varkenvarken/blenderaddons#17]
Slope addon does not work on Blender 4.1
```py
Error: Python: Traceback (most recent call last):
  File "/home/deck/.config/blender/4.1/scripts/addons/slope.py", line 86, in execute
    reference = reference * wmat
                ~~~~~~~~~~^~~~~~
TypeError: Element-wise multiplication: not supported between 'Vector' and 'Matrix' types
```
![Screenshot_20240407_172542](https://github.com/varkenvarken/blenderaddons/assets/127012288/9611731a-7444-4e7d-9072-067a5d6b3a4e)
![Screenshot_20240407_172557](https://github.com/varkenvarken/blenderaddons/assets/127012288/e9b7dd74-e878-4d0c-9c43-3033c7aad8b8)

I also tried Blender 3.6.10 LTS, but the slope button just does not seem to exist.
![Screenshot_20240407_172734](https://github.com/varkenvarken/blenderaddons/assets/127012288/11d266dd-b17e-4082-bbcd-13dd7708d7e2)
![Screenshot_20240407_172802](https://github.com/varkenvarken/blenderaddons/assets/127012288/2df2a3a1-1381-4838-acb0-61bdced4ff9d)

**Comment by varkenvarken** [comment]
thanks for reporting this. indeed, somewhere in the Blender 3.x series the * operator was replaced by the @ operator for matrix multiplication. this should be an easy fix, I will try to reproduce & fix it

**Comment by varkenvarken** [comment]
Fixed. It was quite some work to update this ancient add-on (11 years old, made for Blender 2.6): Not just the matrix multiplication operator but also operator properties are now annotated class variables, vertex colors have an opacity component, brush curves have changed and the way we access an active object now goes via a view layer. 
Anyway, it should work now :smile:

**Comment by EvanDSanders** [comment]
Thank you! 
I used Mirage in Blender 3.6 to make some terrain, but since I updated to 4.1 after I got what I liked from it, I couldn't use the add-on to update the slope after sculpting the land. This will be of great help, since using GeoNodes to make the slope data is really slow (might just be a skill issue). The slope is a little different and inverted from what Mirage does, but that's not an issue, a simple color ramp fixes that. 😄 

 
Edit: Turns out the slowness was from the Distribute Points on Faces node and not the normal sampling thing, it is still slower, but this still yields better looking results anyways!
: Both node groups take less than 30ms, the direct normal takes just over 30ms, sample nearest takes 620ms.
![Screenshot_20240408_182439](https://github.com/varkenvarken/blenderaddons/assets/127012288/27c407ef-c308-4405-b8da-01abdcf04563)
![Screenshot_20240408_182446](https://github.com/varkenvarken/blenderaddons/assets/127012288/b84ef4fd-9ee7-4586-b719-37ed3d8912e7)
![Screenshot_20240408_182452](https://github.com/varkenvarken/blenderaddons/assets/127012288/4b720e23-8b55-4543-9768-1626a0fde554)
![Screenshot_20240408_182458](https://github.com/varkenvarken/blenderaddons/assets/127012288/0fd308d9-6464-4862-9fb6-a1242bfaca99)
